ios/RNTPTDocumentView.m

```diff
@@ -748,6 +748,38 @@ - (void)saveDocumentWithCompletionHandler:(void (^)(NSString * _Nullable filePat
     }];
 }
 
+- (void)setFlagForFields:(NSArray<NSString *> *)fields setFlag:(PTFieldFlag)flag toValue:(BOOL)value
+{
+    PTPDFViewCtrl *pdfViewCtrl = self.documentViewController.pdfViewCtrl;
+    BOOL shouldUnlock = NO;
+    @try {
+        [pdfViewCtrl DocLock:YES];
+        shouldUnlock = YES;
+        
+        PTPDFDoc *doc = [pdfViewCtrl GetDoc];
+
+        for (NSString * fieldName in fields) {
+            PTFieldIterator * fieldItr = [doc GetFieldIterator];
+
+            // loop through full iterator in case multiple fields match name
+            for(; [fieldItr HasNext]; [fieldItr Next]) {
+                PTField * field = [fieldItr Current];
+
+                if([[field GetName] isEqualToString:fieldName]) {
+                    [field SetFlag:(PTFieldFlag)flag value:value];
+                }
+            }
+        }
+    }
+    @finally {
+        if (shouldUnlock) {
+            [pdfViewCtrl DocUnlock];
+        }
+    }
+    
+    [pdfViewCtrl Update:YES];
```

Comment:
The call to `[PTPDFViewCtrl Update:]` should be moved up, to just after the end of `for (NSString * fieldName in fields) { ... }`, inside the doc-locked section.